docs/design/build-system.md

```diff
@@ -17,13 +17,46 @@ Many tools are used in this process, and most of these steps happen within a [We
 We maintain a number of Webpack-centric plugins in this repository, some of these are public but most are private since they are very specific to our setup.
 
 
+## Overview diagram
```

Comment:
Addresses https://github.com/angular/angular-cli/pull/16400#discussion_r358402434 and https://github.com/angular/angular-cli/pull/16400#discussion_r358403869.

Reply:
@clydin @alan-agius4 PTAL as well.

Reply:
I'd recommend using Graphviz to render this graph if possible. If you pull out a dot graph to a separate file, you can use http://www.gravizo.com/ to render it in the markdown.

Reply:
The style sheet pipeline appears to be reversed.  The preprocessors are first with the raw/style loaders last.

Reply:
Yes indeed, also the `scripts array` go through `terser-webpack-plugin` and not `terser`.

Reply:
Ah yes the good old "reverse array order". How could I forget. Again.

Reply:
This graphizo thing is nice, great suggestion @dgp1130. Updated to be a inline image in DOT.

Reply:
Unfortunately the preview doesn't seem to be loading correctly. The one issue I've had with the service is getting it to work without binding to a specific `master` URL. Embedded the graph directly never seems to work for me. I've had better luck putting the DOT graph in a separate file and then linking to that ([example](https://github.com/dgp1130/chatter/#architecture)), though this doesn't work for previews.

Reply:
Well that's unfortunate. It does show up correctly on my VSCode markdown preview:
![image](https://user-images.githubusercontent.com/4172079/72381870-0ddfda00-3710-11ea-80c9-5b740c87a12c.png)

Reply:
Graph LGTM. Only minor nit is that I tend to prefer `node [shape=rectangle];` as I think that generally looks better for most graphs. Up to you if you want to do that.

Reply:
Updated to a separated .dot file.

Reply:
Updated to rectangle shapes, looks like this now:

![image](https://user-images.githubusercontent.com/4172079/72474745-629f5580-37e1-11ea-8ea9-ae337373864e.png)